### PR TITLE
code cleanup: Abstract repetitive codes in `cmd` as a function

### PR DESCRIPTION
**Pull request by binacs**
#### What type of PR is this?

/kind cleanup
/kind design

#### What this PR does / why we need it:

There are many repetitive codes such as:

```go
	usageFmt := "Usage:\n  %s\n"
 	cols, _, _ := term.TerminalSize(cmd.OutOrStdout())
 	cmd.SetUsageFunc(func(cmd *cobra.Command) error {
 		fmt.Fprintf(cmd.OutOrStderr(), usageFmt, cmd.UseLine())
 		cliflag.PrintSections(cmd.OutOrStderr(), namedFlagSets, cols)
 		return nil
 	})
 	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
 		fmt.Fprintf(cmd.OutOrStdout(), "%s\n\n"+usageFmt, cmd.Long, cmd.UseLine())
 		cliflag.PrintSections(cmd.OutOrStdout(), namedFlagSets, cols)
 	})
```

Obviously we can abstrat them into one and make the codes more elegant.

#### Which issue(s) this PR fixes:

Fixes #

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by BinacsLee** [comment]
cc @cheftako @dims 
Thanks a lot!

**Comment by BinacsLee** [comment]
Hi @mikedanese ,
I update the govendor to fix kubernetes-dependencise, could you please review it again?
Thanks a lot!

**Comment by BinacsLee** [comment]
cc @cheftako @dims 
Thanks ^-^

**Comment by dims** [comment]
/lgtm
/approve

/assign @liggitt 
(for any remaining approvals)

File: cmd/kube-scheduler/app/server.go [hunk 1]
```diff
@@ -100,18 +100,8 @@ for more information about scheduling and the kube-scheduler component.`,
...
-	cmd.SetUsageFunc(func(cmd *cobra.Command) error {
-		fmt.Fprintf(cmd.OutOrStderr(), usageFmt, cmd.UseLine())
-		cliflag.PrintSections(cmd.OutOrStderr(), namedFlagSets, cols)
-		return nil
-	})
-	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
-		fmt.Fprintf(cmd.OutOrStdout(), "%s\n\n"+usageFmt, cmd.Long, cmd.UseLine())
-		cliflag.PrintSections(cmd.OutOrStdout(), namedFlagSets, cols)
-	})
-	cmd.MarkFlagFilename("config", "yaml", "yml", "json")
```
Comment (liggitt) [inline cmd/kube-scheduler/app/server.go:114]: the `MarkFlagFilename` line was dropped accidentally

**Comment by BinacsLee** [comment]
@liggitt 
Sorry, it's my mistake.
I have fixed it, and update govendor based on the latest commit, could you please review it again?
Thank you very much!

File: go.mod [hunk 2]
```diff
@@ -42,6 +42,7 @@ require (
 	github.com/evanphx/json-patch v4.9.0+incompatible
 	github.com/fsnotify/fsnotify v1.4.9
 	github.com/go-bindata/go-bindata v3.1.1+incompatible
+	github.com/go-openapi/spec v0.19.5
```
Comment (liggitt) [inline go.mod:45]: this is surprising... is it correct?

Comment (BinacsLee) [reply go.mod]: Modified by `hack/update-vendor.sh`, I had fixed it.

**Comment by BinacsLee** [comment]
@liggitt  I'm very sorry for your time.
I force-pushed the branch because of the test error, now `lgtm` is needed.
Forgive me for my inexperience, I will avoid these problems from the beginning in the future.

**Pull request merged**